src/compiler/checker.ts

```diff
@@ -923,6 +923,14 @@ namespace ts {
             return combined;
         }
 
+        /** CommonJS is *almost* a module -- no merges -- except that assignment declarations need to merge with globals. */
+        function mergeCommonJsSymbolTable(target: SymbolTable, source: SymbolTable) {
+            source.forEach((sourceSymbol, id) => {
+                if (target.has(id) && sourceSymbol.flags & SymbolFlags.Assignment) {

```

Comment:
Talking with @weswigham, he pointed out that it would be better to store all the identifier-unbound declarations in a separate per-file commonJSGlobalAugmentations property that gets merged into globals instead of trying to filter here (there are plenty of symbols marked with Assignment that shouldn't necessarily merge with globals).